src/realm/alloc_slab.cpp

```diff
@@ -338,6 +338,10 @@ MemRef SlabAlloc::do_alloc(const size_t size)
     // Round upwards to nearest page size
     new_size = ((new_size - 1) | (page_size() - 1)) + 1;
 
+    if (ref + new_size > maximum_realm_size) {
+        throw std::runtime_error("Realm size exceeded " + realm::util::to_string(maximum_realm_size) + " bytes");
```

Comment:
or "The Realm file cannot grow larger than the current size of ... bytes."

How is this expected to be handled by SDK's?